src/Symfony/Component/BrowserKit/Tests/ClientTest.php

```diff
@@ -48,6 +51,11 @@ protected function doRequest($request)
         return $response;
     }
 
+    public function getRequest()
```

Comment:
where is this method (and corresponding property) used?

Reply:
It’s used in the tests to check that the header has been added

Reply:
```
        $this->assertEquals($client->getRequest()->getServer()['HTTP_X_REQUESTED_WITH'], 'XMLHttpRequest');
```

Reply:
I'm sorry I don't understand. I tested locally after removing all the changes don here to the `TestClient` class and tests still do pass.

Reply:
Ahh @nicolas-grekas is right. The `Client` already keeps the request internally :blush: 

https://github.com/Simperfit/symfony/blob/00f26e280c5aec7c27fd25c825285546887a1771/src/Symfony/Component/BrowserKit/Client.php#L36

So the changes to `TestClient` are indeed not needed and should be reverted.